### PR TITLE
add vignette and pathway enrichment function

### PR DESCRIPTION
**Pull request by faye-yang**

### CODE REVIEW

File: R/pathwayEnrich.R [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+#'  
+#' @param method method of cutoff that the visualization plot method can be BH or 
+#' bonferroni.
+#' reactInfo
+#' @param reactInfo a dataframe map the reactome pathway id to the reference gene symbol 
+#' 
+#' @return a barplot that show the enriched pathway and its p value(fisher test) 
+#' that is smaller than the ajusted p value.
+#' 
+#' @import ggplot2
```
Comment (rn1n3r) [inline R/pathwayEnrich.R:17]: Shouldn't need import statements here, can be handled in the DESCRIPTION file.

File: R/pathwayEnrich.R [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  for(gene in geneSet){
+    pws<-reactInfo$REACTOME_ID[reactInfo$HGNC==gene]
+    for(pw in pws){
+      if(!is.na(pw)){
+        occurenceGene[[pw]]<-occurenceGene[[pw]]+1
+      }
+      
+    }
+  }
+  enrichment<-data.frame(pathway=c(),pval=c(),adjustedPvalBon=c())
```
Comment (rn1n3r) [inline R/pathwayEnrich.R]: I would personally add some newlines to between code and the next comment line for readability

File: DESCRIPTION [hunk 3]
```diff
@@ -7,11 +7,10 @@ Authors@R: c(
...
                     comment = c(ORCID = "0000-0002-1134-6758")),
-             person("Rachel", "Silverstein", 
-                    email = "rachel.silverstein@mail.utoronto.ca", 
-                    role = c("ctb"), 
-                    comment = c(ORCID = "0000-0001-5724-2252"))
-    )
+              person("Yufei", "Yang", 
+                    email = "faye.yang@utoronto.ca", 
+                    role = c( "ctb"), 
+                    comment = c(ORCID = "")))
```
Comment (rn1n3r) [inline DESCRIPTION]: ORCID missing

File: R/pathwayEnrich.R [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+#' 
+#' @import ggplot2
+#' @importFrom stats fisher.test
+#' @importFrom stats p.adjust
+#' @author {Yufei Yang} (aut)
+#' @examples
+#' # Picking sample gene list to find the pathway enrichment and p value
+#' # Call the symComp helper function to generate list
+#' sampleGeneList<-c("CREBBP","NR1H3","RORA","SREBF1","SMARCD3")
+#' reactInfo<- fetchData("ReactomeSym")
```
Comment (rn1n3r) [inline R/pathwayEnrich.R]: Is this ever not with the ReactomeSym data? If not you could consider just leaving that parameter out and running fetchData() inside the function

**Review by rn1n3r — DISMISSED** [review]

File: R/pathwayEnrich.R (hunk 4 above)
Comment (faye-yang) [reply R/pathwayEnrich.R]: ReactomeSym is newly added data from fetchData()

File: DESCRIPTION (hunk 3 above)
Comment (faye-yang) [reply DESCRIPTION]: How can I find an ORCID?

File: R/pathwayEnrich.R (hunk 1 above)
Comment (faye-yang) [reply R/pathwayEnrich.R:17]: Thank you so much. I will be careful about that.

File: DESCRIPTION (hunk 3 above)
Comment (faye-yang) [reply DESCRIPTION]: I found the id. should I update it now?

File: R/pathwayEnrich.R (hunk 1 above)
Comment (faye-yang) [reply R/pathwayEnrich.R:17]: Do you mean I should delete the import?

File: R/pathwayEnrich.R (hunk 4 above)
Comment (faye-yang) [reply R/pathwayEnrich.R]: sample gene list and fetchData() are both needed reactomeSym just mapping the Reactome id to the gene symbol.

File: R/pathwayEnrich.R (hunk 1 above)
Comment (faye-yang) [reply R/pathwayEnrich.R:17]: I think imports are needed because the r check cannot pass without them.

File: DESCRIPTION [hunk 5]
```diff
@@ -7,11 +7,10 @@ Authors@R: c(
                     email = "boris.steipe@utoronto.ca", 
                     role = c("aut", "cre"), 
                     comment = c(ORCID = "0000-0002-1134-6758")),
-             person("Rachel", "Silverstein",
```
Comment (hyginn) [inline DESCRIPTION]: This is actually a merge conflict - and I don't know why it was not flagged as such. Did you delete Rachel's entry? Anyway: update DESCRIPTION to the latest version before we can check this in.

File: R/pathwayEnrich.R [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    for(pw in pws){
+      if(!is.na(pw)){
+        pwList<-c(pwList,pw)
+      }
+      
+    }
+  }
+  
+  uniquePW<-unique(pwList)
+  #occurance list of pathway for given gene set initialized
```
Comment (hyginn) [inline R/pathwayEnrich.R]: typo: *occurrence

File: R/pathwayEnrich.R [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    occurSet<-occurenceGene[[pathway]]
+    #fisher matrix column for the given gene set
+    #how many pws are in system
+    fisher_matrix[1,1]<-occurSet
+    #how many time this pw are not occur in system
+    fisher_matrix[2,1]<-length(pwList)-occurSet
+    pval<-fisher.test(fisher_matrix,workspace = 200000,alternative = "two.side")$p.value
+    temp<-data.frame(pathway=pathway,pval=pval)
+    enrichment<-rbind(enrichment,temp)
+  }
```
Comment (hyginn) [inline R/pathwayEnrich.R:90]: Coding style: `return()` values must be explicit.

File: R/pathwayEnrich.R [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+  
+  #sort by p value get the p value that is larger than 0.05
+  #since the null hypothesis is the pathway is enriched
+  enrichment<-enrichment[order(enrichment$pval),]
```
Comment (hyginn) [inline R/pathwayEnrich.R]: coding style: spaces

File: vignettes/pathwayEnrichment.Rmd [hunk 9]
```diff
@@ -0,0 +1,116 @@
+---
```
Comment (hyginn) [inline vignettes/pathwayEnrichment.Rmd:1]: Many issues here ...  but none that will prevent merging.

**Review by hyginn — CHANGES_REQUESTED** [review]

File: R/pathwayEnrich.R (hunk 7 above)
Comment (faye-yang) [reply R/pathwayEnrich.R:90]: change to return dataframe and plot ggplot in the vignette

File: R/pathwayEnrich.R (hunk 6 above)
Comment (faye-yang) [reply R/pathwayEnrich.R]: fixed typo

File: R/pathwayEnrich.R (hunk 8 above)
Comment (faye-yang) [reply R/pathwayEnrich.R]: fixed space

File: DESCRIPTION (hunk 5 above)
Comment (faye-yang) [reply DESCRIPTION]: resolved conflict

**Comment by hyginn** [comment]
You were instructed not to push further commits. Closing.

**Review by hyginn — APPROVED** [review]
Some details still need to be addressed, but most reviewers' concerns were resolved. Approach is sound, looks like valid starting point for enrichment analysis in ESA. Since later changes were not sync'd will merge manually.

**Comment by hyginn** [comment]
Will merge manually. Closed.

**Pull request closed without merging**